fdbcli/BulkLoadCommand.actor.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * BulkLoadCommand.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbcli/fdbcli.actor.h"
+
+#include "fdbclient/IClientApi.h"
+
+#include "fdbclient/ManagementAPI.actor.h"
+#include "fdbclient/BulkLoading.h"
+
+#include "flow/Arena.h"
+#include "flow/FastRef.h"
+#include "flow/ThreadHelper.actor.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+namespace fdb_cli {
+
+ACTOR Future<UID> bulkLoadCommandActor(Reference<IClusterConnectionRecord> clusterFile,
+                                       Database localDb,
+                                       std::vector<StringRef> tokens) {
+	if (tokencmp(tokens[1], "mode")) {
+		if (tokens.size() != 3) { // TODO(Zhe): check by ID
+			printUsage(tokens[0]);
+			return UID();
+		}
+		if (tokencmp(tokens[2], "on")) {
+			int old_ = wait(setBulkLoadMode(localDb, 1));
+			return UID();
+		} else if (tokencmp(tokens[2], "off")) {
+			int old_ = wait(setBulkLoadMode(localDb, 0));
+			return UID();
+		} else {
+			printUsage(tokens[0]);
+			return UID();
+		}
+
+	} else if (tokencmp(tokens[1], "localtask")) {
+		if (tokens.size() < 7) {
+			printUsage(tokens[0]);
+			return UID();
+		}
+		Key rangeBegin = tokens[2];
+		Key rangeEnd = tokens[3];
+		if (rangeBegin > normalKeys.end || rangeEnd > normalKeys.end) {
+			printUsage(tokens[0]);
+			return UID();
+		}
```

Comment:
nit: Check endkey > startkey?

Reply:
We submit task w/o any checking of the SST or byte sample files?

Reply:
> Check endkey > startkey?
This is checked when building KeyRangeRef.

> We submit task w/o any checking of the SST or byte sample files?
I feel it is too heavy for FDB client to check the files but the FDB server should check the files when injecting them. Also client should check the file before using FDB cli bulk loading.

Reply:
Ok but I'd advocate running basic verifications before copying/taking ownership of the files. An exception thrown at this point before ingest would be in the user's face; one later thrown by server will be harder to dig up.